src/system/SystemObject.h

```diff
@@ -123,28 +123,27 @@ class DLL_EXPORT Object
     std::mutex * mMutexRef;
 #endif
 
-    Layer * volatile mSystemLayer; /**< Pointer to the layer object that owns this object. */
-    unsigned int mRefCount;        /**< Count of remaining calls to Release before object is dead. */
+    unsigned int mRefCount; /**< Count of remaining calls to Release before object is dead. */
 
-    /** If not already retained, attempt initial retention of this object for \c aLayer and zero up to \c aOctets. */
-    bool TryCreate(Layer & aLayer, size_t aOctets);
+    /** If not already retained, attempt initial retention of this object and zero up to \c aOctets. */
```

Comment:
The comment does not seem as clear (we always zero aOctets, not only up to), zero-ing follows sizeof(this).

Consider:

```cpp
/// Attempts to perform an initial retention of this object.
/// If reference count is non-zero, tryCreate will fail and return false.
/// If reference count is zero, then:
///    - reference count will be set to  1
///    - The size of the created object (assumed to be derived from SystemObject) is aOctects.
///      the method will memset to 0 the bytes following sizeof(SystemObject). 
///
/// Typical usage is like:
///    class Foo: public SystemObject {...}
///    ....
///    Foo foo;
///    foo.TryCreate(sizeof(foo));
///
///  IMPORTANT inheritance precondition:
///      0 memset assumes that SystemObject is the top of the inheritance. This will NOT
///  work properly: 
///      class Bar: public Baz, SystemObject {...}
///      Bar bar;
///      bar.TryCreate(sizeof(bar)); /// NOT safe: this will clear Baz extra bytes in unallocated space.
```

Reply:
Acked